src/components/cylc/tree/cylc-tree.js

```diff
@@ -86,6 +86,25 @@ function computeCyclePointsStates (cyclePointNodes) {
   }
 }
 
+function sortTaskProxyOrFamilyProxy (left, right) {
+  // sort cycle point children (family-proxies, and task-proxies)
+  // first we sort by type ascending, so 'family-proxy' types come before 'task-proxy'
+  // then we sort by node name ascending, so 'bar' comes before 'foo'
+  // node type
+  if (left.type < right.type) {
+    return -1
+  }
+  if (left.type > right.type) {
+    return 1
+  }
+  // name
+  return left.node.name.toLowerCase()
+    .localeCompare(
+      right.node.name.toLowerCase(),
+      undefined,
+      { numeric: true, sensitivity: 'base' })
+}
+
```

Comment:
We cannot use the binary-search function, as it will simply search by one value. Tasks and Family Proxies are grouped together, so the location of the next insertion depends on type and value.